### ISSUE TITLE
Provide a path for external gps sensors to report accuracy

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#4768]
As of Collect v1.30, the accuracy of points coming from the mock location provider is always submitted as 0. This was intended to identify points coming from location mocking apps. It turns out that external location sensors provide location through such apps. As far as I know, there’s no way to differentiate between the two but that would be a good thing to confirm.
https://forum.getodk.org/t/show-accuracy-for-gps-mock-location-for-external-sensors/34455/2

The discussion on exactly what to do should happen on the forum. This issue is a placeholder so we make sure it’s addressed.

**Comment by seadowg** [comment]
Ok to solve this we're going to introduce an XLSForm parameter for geo questions (`geopoint`, `geotrace` and `geoshape`):

| type | name | label | parameters |
|------------ |------------ |------------ |-----------------------| 
| geopoint | location | Location | allow-mock-accuracy=true |

When the `allow-mock-accuracy` parameter is `true`, the client should allow “mock” locations. In Collect, this will mean that locations from Android’s mock providers will use the reported accuracy. The default for `allow-mock-accuracy` will be `false` so this behaviour will be “opt-in” and by default, “mock” locations will still have an accuracy of 0. This parameter will be represented in an XForm as a bind attribute `odk:allow-mock-accuracy` for the relevant data types (which will require changes to pyxform).

**Comment by yanokwa** [comment]
The alternative I had suggested to @seadowg is that we revert the behavior from the previous release and use whatever accuracy the mock provider reports. Then we add a parameter of `allow-mock-providers` and default that to `true.` And if a manager sets it to `false`, we prevent anyone from using mock providers entirely.

@lognaturel's pushback about my alternative is that: 
> most people don’t want mock providers, don’t want data from them, and will never know that they are being used. That was why the issue was filed in the first place. Setting the accuracy to 0 means there’s always a marker of something going wrong for the common case.

@seadowg was also concerned:
> Let me think on it. I’m also not sure about our UI support for something like that. I think right now if you were banning mock locations you would end up with an infinite spinner on the geopoint dialog.

I discussed this with @lognaturel and I'm OK proceeding. To summarize...

With this implementation, if an enumerator uses a mock provider, we set accuracy to 0 so managers know there might be fake data. If they want to change this behavior, they can use `allow-mock-accuracy=true`. If they want to prevent data collectors from using mock providers, they can use a constraint of `selected-at(.,3) > 0`
